python/pyspark/sql/connect/streaming/query.py

```diff
@@ -230,21 +233,26 @@ def resetTerminated(self) -> None:
 
     resetTerminated.__doc__ = PySparkStreamingQueryManager.resetTerminated.__doc__
 
-    def addListener(self, listener: Any) -> None:
-        # TODO(SPARK-42941): Change listener type to Connect StreamingQueryListener
-        # and implement below
-        raise NotImplementedError("addListener() is not implemented.")
+    def addListener(self, listener: StreamingQueryListener) -> None:
+        listener._init_listener_id()
+        cmd = pb2.StreamingQueryManagerCommand()
+        expr = proto.PythonUDF()
+        expr.command = CloudPickleSerializer().dumps(listener)
+        expr.python_ver = "%d.%d" % sys.version_info[:2]
+        cmd.add_listener.python_listener_payload.CopyFrom(expr)
+        cmd.add_listener.id = listener._id
+        self._execute_streaming_query_manager_cmd(cmd)
+        return None
```

Comment:
can be removed - all of `return None`.

Reply:
Actually I do remember before I tried to remove these `return None`s but in the end it fails some python lint test

Reply:
https://github.com/apache/spark/pull/40785#issuecomment-1515737988

Reply:
We can omit `return None` (or `return`) when the return type is `None`.